data/option_test.go

```diff
@@ -0,0 +1,65 @@
+package data_test
+
+import (
+	"GoSlice/data"
+	"GoSlice/util/test"
+	"strings"
+	"testing"
+)
+
+func TestFanSpeedDefaults(t *testing.T) {
+	opts := data.DefaultOptions()
+	test.Equals(t, opts.Filament.FanSpeed.LayerToSpeedLUT[2], 255)
+}
+
+func TestFanSpeedNegative(t *testing.T) {
```

Comment:
I like the table driven tests often used in go:
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests
This allows very easy adding of many similar test cases.
What do you think about it?

So I would suggest something like this:
```
// layerToSpeedLUTComparer returns a cmp.Comparer which can handle data.FanSpeedOptions.
func fanSpeedOptionsComparer() cmp.Option {
	return cmp.Comparer(func(p1, p2 data.FanSpeedOptions) bool {
		for layer, speed := range p1.LayerToSpeedLUT {
			if p2.LayerToSpeedLUT[layer] != speed {
				return false
			}
		}

		return true
	})
}

func TestSetFanSpeed(t *testing.T) {
	var testCases = map[string]struct {
		optionString   string
		expectedError string
		expected *data.FanSpeedOptions
	}{
		"FanSpeedNegative": {
			optionString: "1=-5",
			expectedError: "fan control needs to be in format",
			expected: nil,
		},
		"FanSpeedOneSuccessful": {
			optionString: "1=10",
			expectedError: "",
			expected: &data.FanSpeedOptions{LayerToSpeedLUT: map[int]int{
				1: 10,
			}},
		},
	}

	for testName, testCase := range testCases {
		t.Log("testCase:", testName)
		actual := data.FanSpeedOptions{}
		err := actual.Set(testCase.optionString)
		if testCase.expectedError != "" {
			test.Assert(t, strings.Contains(err.Error(), testCase.expectedError), "error expected")
			return
		}

		test.Equals(t, testCase.expected, actual, fanSpeedOptionsComparer())

	}
}
```

Reply:
done